waspc/waspls/test/Wasp/LSP/DebouncerTest.hs

```diff
@@ -0,0 +1,54 @@
+module Wasp.LSP.DebouncerTest
+  ( spec_Debouncer,
+  )
+where
+
+import Control.Concurrent (newEmptyMVar, threadDelay, tryPutMVar, tryReadMVar)
+import Control.Monad (replicateM_, void)
+import GHC.Conc (atomically, newTVarIO, readTVar, readTVarIO, writeTVar)
+import Test.Tasty.Hspec
+import Wasp.LSP.Debouncer (debounce, newDebouncerIO)
+
+spec_Debouncer :: Spec
+spec_Debouncer = describe "Wasp.LSP.Debouncer" $ do
+  it "runs the action" $ do
+    debouncer <- newDebouncerIO
+    mvar <- newEmptyMVar
+
+    debounce debouncer 1000 () (void $ tryPutMVar mvar ())
+    threadDelay 20000
```

Comment:
Does this mean that our unit tests have a minimum time of running of 20 seconds? 😄

Reply:
Thankfully not! `threadDelay` takes a delay in microseconds, so the longest test is only 40ms.